src/vest.ts

```diff
@@ -0,0 +1,48 @@
+import { FieldValues, Resolver, transformToNestObject } from 'react-hook-form';
+
+type VestErrors = Record<string, string[]>;
+
+const parseErrorSchema = (
+  vestError: VestErrors,
+  validateAllFieldCriteria: boolean,
+) => {
+  return Object.entries(vestError).reduce((prev, [key, value]) => {
+    return {
+      ...prev,
+      [key]: {
+        type: '',
+        message: value[0],
+        ...(validateAllFieldCriteria
+          ? {
+              types: value.reduce((prev, message, index) => {
+                return {
+                  ...prev,
+                  [index]: message,
+                };
+              }, {}),
+            }
+          : {}),
+      },
+    };
+  }, {});
+};
+
+export const vestResolver = <TFieldValues extends FieldValues>(
+  schema: any,
```

Comment:
A schema _(validationSuite)_ is typed as `ICreateResult` but it not exposed, you can retrieve it by doing the following thing: 

```ts
type ICreateResult = ReturnType<typeof vest.create>;
```

Reply:
great suggestion